Umee-Bots/Large-Borrow/src/agent.ts

```diff
@@ -0,0 +1,48 @@
+import { ethers, Finding, getEthersProvider, HandleTransaction, TransactionEvent } from "forta-agent";
+import BigNumber from "bignumber.js";
+import { BALANCE_OF_ABI, BORROW_ABI, GET_RESERVE_DATA_ABI } from "./constants";
+import { AgentConfig, createFinding, SmartCaller } from "./utils";
+import CONFIG from "./agent.config";
+
+BigNumber.set({ DECIMAL_PLACES: 18 });
+
+export const provideHandleTransaction = (
+  provider: ethers.providers.Provider,
+  config: AgentConfig
+): HandleTransaction => {
+  const LendingPool = SmartCaller.from(
+    new ethers.Contract(config.lendingPoolAddress, [GET_RESERVE_DATA_ABI], provider),
+    {
+      cacheByBlockTag: false,
+    }
+  );
+  const threshold = new BigNumber(config.tvlPercentageThreshold);
+
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const logs = txEvent.filterLog(BORROW_ABI, config.lendingPoolAddress);
+
+    await Promise.all(
+      logs.map(async (log) => {
+        const { reserve, amount, user, onBehalfOf } = log.args;
+
+        const { uTokenAddress } = await LendingPool.getReserveData(reserve, { blockTag: "latest" });
```

Comment:
why `"latest"` here?

Reply:
Well, it shouldn't matter a lot here to use the block number or `"latest"`, because this shouldn't change (internally, `SmartCall` should be frequently getting it from the cache).

Reply:
Ok, because lendingPool is not caching by blockTag it is being cached. :+1: